### PR TITLE
feat: add DingTalk(钉钉) tool for sending message to chat group bot via webhook

### PR DESCRIPTION
**Pull request by bowenliang123**
# Description
- prepare the group bot in DingTalk group and getting the access_token from webhook URL and the secret key in security setting.
image

- config dingtalk tool in app
Pasted Graphic 2

Pasted Graphic

- result with calling dingtalk_group_bot
Pasted Graphic 1

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  icon: icon.svg
+description:
+  human:
+    en_US: Sending a group message on DingTalk via the webhook of group bot
+    zh_Hans: 通过钉钉的群机器人webhook发送群消息
+    pt_BR: Sending a group message on DingTalk via the webhook of group bot
+  llm: A tool for sending messages to a chat group on DingTalk(钉钉) .
+parameters:
+  - name: access_token
+    type: string
```
Comment (crazywoola) [inline api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.yaml]: Please use this type `type: secret-input`.

File: api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    label:
+      en_US: access token
+      zh_Hans: access token
+      pt_BR: access token
+    human_description:
+      en_US: access_token in the group robot webhook
+      zh_Hans: 群自定义机器人webhook中access_token字段的值
+      pt_BR: access_token in the group robot webhook
+    form: form
+  - name: secret
```
Comment (crazywoola) [inline api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.yaml]: Please use this type `type: secret-input`.

File: api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                ) -> Union[ToolInvokeMessage, list[ToolInvokeMessage]]:
+        """
+            invoke tools
+            Dingtalk group chat api: https://open.dingtalk.com/document/orgapp/custom-robot-access
+        """
+        content = tool_parameters.get('content', '')
+        if not content:
+            return self.create_text_message('Invalid parameter content')
+
+        access_token = tool_parameters.get('access_token', '')
```
Comment (crazywoola) [inline api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.py]: And use this to get sk `self.runtime.credentials["access_token"]`

**Review by crazywoola — CHANGES_REQUESTED** [review]

File: api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.yaml (hunk 1 above)
Comment (bowenliang123) [reply api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.yaml]: Thanks. Resolved.

File: api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.py (hunk 3 above)
Comment (bowenliang123) [reply api/core/tools/provider/builtin/dingtalk/tools/dingtalk_group_bot.py]: Thanks. Resolved.

**Comment by bowenliang123** [comment]
Will fix the errors with `select-secret` type in this PR soon.

**Comment by Yeuoly** [comment]
> Will fix the errors with `select-secret` type in this PR soon.

secret-input is not supported yet in tool parameters, and all the credentials should not be stored in parameters but in provider credentials.

We are cleared that multi credential configurations are necessary for several scenarios including this one, and it's coming soon.

**Review by Yeuoly — APPROVED** [review]

**Comment by Yeuoly** [comment]
LGTM, already tested locally.

**Review by takatost — APPROVED** [review]

**Pull request merged**